impl/internal/did/did.go

```diff
@@ -792,3 +797,19 @@ func keyTypeForJWK(jwk jwx.PublicKeyJWK) int {
 	}
 	return -1
 }
+
+// chunkTextRecord splits a text record into chunks of 255 characters
+func chunkTextRecord(record string) []string {
+	var chunks []string
+	for len(record) > 255 {
+		chunks = append(chunks, record[:255])
+		record = record[255:]
+	}
```

Comment:
TIL (thanks chad!) that `len(str)` returns the number of bytes. This may not be equivalent to the number of chars. For instance, `len("£")` returns `2`.

Given the above, if you want to actually do this into 255 chars, then see the function below.

```go
// SplitString255 splits a string into substrings of up to 255 characters.
func SplitString255(input string) []string {
	var chunks []string
	runeArray := []rune(input) // Convert to rune slice to properly handle multi-byte characters

	for len(runeArray) > 0 {
		if len(runeArray) <= 255 {
			chunks = append(chunks, string(runeArray))
			break
		}

		chunks = append(chunks, string(runeArray[:255]))
		runeArray = runeArray[255:]
	}

	return chunks
}
```

But the real question is whether the library should allow setting a value in a DID Document which contains values outside the ASCII range. I think the library should. And thus, I believe the value should be base64url encoded before doing chunking, so there are no issues with existing DNS systems. If you agree, then my suggestion doesn't matter, as all ASCII chars are 1-byte when encoded in UTF-8.

Reply:
this is a great catch about bytes vs character sizes - thank you
I am opposed to b64 encoding because of [size increases](https://bharatkalluri.com/posts/base64-size-increase-explanation)

Reply:
I am going to merge, but please feel free to open an issue to support non-ASCII characters.